### ISSUE TITLE
Smooth scrolling is bad if you hold down `j`

### ISSUE THREAD

**Issue by bovine3dom** [tridactyl/tridactyl#1625]
Related to #627.

Steps to reproduce:

1. Visit a long web page e.g. https://en.wikipedia.org/wiki/Battle_of_Mosul_(2016%E2%80%932017)
2. `set smoothscroll true`
3. Scroll down by tapping `j` a few times, observe that it's quite good (as far as smooth scrolling can be)
4. Hold down `j` - it's jerky.

Expected behaviour: scrolling should be smooth. Possible fixes: have a lock for scrolling so that multiple scrolls can't be initiated at once; support keyup/keydown events in the parsers.

**Comment by diogox** [comment]
Just my two cents on this. 

I was using Vimium on Firefox before moving to tridactyl, which is much better on most things btw :wink:, (Thanks for making this great extension!), and the smooth scrolling there was blissful (at least for me). It doesn't have the 'hold down `j`' problem either.

Anyway, if you decide to take a look at their code to get some ideas, here's the link to their [scroller script](https://github.com/philc/vimium/blob/8c9404f84fd4310b89818e132c0ef0822cbcd059/content_scripts/scroller.coffee). It's in coffescript though.

**Comment by glacambre** [comment]
Things might have changed since last year but last time I took a look, vimium wasn't able to scroll on as many websites as tridactyl (see https://github.com/tridactyl/tridactyl/issues/627#issuecomment-412251658 ).

**Comment by diogox** [comment]
I did notice `tridactyl` scrolls on more sites. I was having trouble with gmail on `vimium`. Escaping didn't help either.

I see this already has had some discussion around it. I do, however agree with the [guy that replied to the comment you sent](https://github.com/tridactyl/tridactyl/issues/627#issuecomment-485150142). 
The issue seems to have been closed, but `vimium`'s smooth scrolling is still better imo. Not sure why that is, didn't really look at the code that much.
Not to be ungrateful, but this is probably the one thing I wish got fixed. Everything else seems perfect so far :+1: 

Btw, from the little time I spent looking at the code, regarding the jerky scrolling when holding down `j`, it seems like `vimium` just disregards the new 'scroll down' events, until it detects the `key up` from the first event that was sent. Not sure if that's useful, but I thought I'd just leave it here, just in case...
